### PR TITLE
test(/v1/search): Added test case for /v1/search API with Mockito

### PR DESCRIPTION
**Pull request by shangan23**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/web-slate/simple-google-search/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What type of change does this PR introduce?

- [x] API
- [ ] Frontend

## PR Sub Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe: Test case for API

## What is the current behavior?

Issue Number: #32 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Step 1:
Create Test for Search Controller - `SearchControllerTest.java`
Create Test for searchResultv1Test method - `searchResultv1Test`

Step 2:
Add New Class `SearchControllerTest` with below decorators.

- @RunWith(SpringRunner.class) -  tells JUnit to run using Spring's testing support. 
- @SpringBootTest - tells Spring Boot to look for the main configuration class
- @AutoConfigureMockMvc - Annotation that can be applied to a test class to enable and configure auto-configuration of MockMvc
- @MockBean - to add mock objects to the Spring application context.
- @Test

Step 3:
Add New Test Case method
```java
void searchResultv1Test() throws Exception {
    // These steps generate the response which needs to be compared with the mock API hit below.
    String response = "{\"total_results\":6,\"execution_time\":45000,\"query_string\":\"facebook\",\"results\":[{...}]}";
    when(searchController.searchResultv1("facebook")).thenReturn(response);
    
    // These steps produce the response from mock API and compare to the above-generated response.
    mockMvc.perform(get("/v1/search").param("query","facebook").content(MediaType.APPLICATION_JSON_VALUE))
    .andExpect(jsonPath("$.query_string").exists())
    .andExpect(jsonPath("$.results").exists())
    .andExpect(jsonPath("$.query_string").value("facebook"))
    .andExpect(status().isOk())
    .andReturn();
}
```
Step 4:
How to run the test case in command?
Working on Installing Maven on MacOs - #57 to utilize in the command line.
- On Eclipse -> Right click on Project -> Run As -> JUnit Test

### CODE REVIEW

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	
+	@MockBean
+	private SearchController searchController;
+	
+	@Before
+	public void setUp() {
+		mockMvc = MockMvcBuilders.webAppContextSetup(context).build();
+	}
+
+	@Test
```
Comment (ramsunvtech) [inline api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:40]: In next PR, would like to see Test Case title.

**Comment by ramsunvtech** [comment]
@shangan23, Awesome start!! very important PR

one small thing. please update the GitHub Template `Other information` to `Steps to Implement / Fix` and add below steps.

Rough Sample
```
Step 1:
Create Test for Search Controller - `SearchControllerTest.java`

Step 2:
Add New Class `SearchControllerTest` with below decorators.
@RunWith(SpringRunner.class) - ????
@SpringBootTest - Use for ??
@AutoConfigureMockMvc - Use for ???

Step 3:
Add New Test Case
void searchResultv1Test() throws Exception {
}
```

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package com.google.search.googlesearch.controller.v1.api;
+
+import static org.mockito.Mockito.when;
```
Comment (ramsunvtech) [inline api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:3]: is mockito added in pom.xml file??

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		when(searchController.searchResultv1("facebook")).thenReturn(response);
+		
+		mockMvc.perform(get("/v1/search").param("query","facebook").content(MediaType.APPLICATION_JSON_VALUE))
```
Comment (ramsunvtech) [inline api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:45]: Thanks i will add remaining test case. #43

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java (hunk 2 above)
Comment (shangan23) [reply api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:3]: no change required in pom.xml

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java (hunk 1 above)
Comment (shangan23) [reply api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:40]: Not sure about it. 
@jpukg - your comments

**Comment by ramsunvtech** [comment]
@shangan23 Looking forward on Steps 👍

**Review by karthikeyanpa12 — APPROVED** [review]

**Review by naveenvignesh5 — APPROVED** [review]

**Review by ramsunvtech — APPROVED** [review]

File: api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java (hunk 1 above)
Comment (jpukg) [reply api/google-search/src/test/java/com/google/search/googlesearch/controller/v1/api/SearchControllerTest.java:40]: @shangan23 exactly done as required.

**Review by jpukg — APPROVED** [review]

**Pull request merged**